docroot/modules/custom/foia_webform/src/AgencyLookupServiceInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Drupal\foia_webform;
+
+use Drupal\node\NodeInterface;
+
+/**
+ * Provides interface defining the Agency Lookup Service.
+ */
+interface AgencyLookupServiceInterface {
+
+  /**
+   * Looks up the Agency Component based on the associated webform.
+   *
+   * @param string $webformId
+   *   The form ID.
+   *
+   * @return \Drupal\node\NodeInterface|null
+   *   The Agency Component object or NULL.
+   */
+  public function getComponentFromWebform($webformId);
+
+  /**
+   * Look up the Agency taxonomy term for a given Agency Component.
+   *
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   *
+   * @return \Drupal\taxonomy\TermInterface
+   *   Returns the Agency taxonomy term object.
```

Comment:
If we update this method to also return `NULL` in case the component isn't associated with an agency for whatever reason, this doc block will need updating as well to indicate that null is a valid return type, similar to what you did for the `getComponentFromWebform` method above.